.gitlab-ci.yml

```diff
@@ -56,18 +56,31 @@ validate-log-integrations:
 trigger-agent-build:
   stage: trigger
   image: $VALIDATE_AGENT_BUILD
+  # Exclude release branches
+  workflow:
+    rules:
+      - if: ($CI_COMMIT_BRANCH !~ /^7.[0-9]{2}.x/)
+        changes:
+          - datadog_checks_base/datadog_checks/base/data/agent_requirements.in
+        when: never
+      - when: on_success
   cache:
     <<: *slack-cache
   script:
     - export SLACK_CACHE_DIR="${PWD}/.slack-cache"
     - DATADOG_AGENT_PIPELINE_URL=$(aws ssm get-parameter --region us-east-1 --name ci.integrations-core.datadog-agent-pipeline-url --with-decryption --query "Parameter.Value" --out text)
     - export AGENT_REQUIREMENTS_CHANGED="false"
     - git fetch origin master
-    - git --no-pager diff --exit-code --name-only HEAD $(git merge-base HEAD FETCH_HEAD) -- datadog_checks_base/datadog_checks/base/data/agent_requirements.in || export AGENT_REQUIREMENTS_CHANGED="true" || true
+    - BASE_REF=$(git merge-base HEAD FETCH_HEAD)
```

Comment:
Latest common commit between HEAD and master. If the latest common commit has tag `7.YY.x-rc.1`, we're based on the release branch and we should trigger the agent build against `datadog-agent/7.YY.x` (when needed), if not, we're based on master and we should trigger the agent build against `datadog-agent/main`